Classes/Sizes/Rootline.php

```diff
@@ -38,145 +36,111 @@ final class Rootline
 
     private BackendLayout $backendLayout;
 
-    private array $rootline = [];
+    private ContainerRepository $containerRepository;
 
     private array $finalSizes = [];
 
     private string $fieldName;
 
-    public function __construct(array $data, string $fieldName)
-    {
+    private string $backendLayoutIdentifier;
+
+    public function __construct(
+        array $data,
+        string $fieldName
+    ) {
```

Comment:
Please start with turning this class in a proper factory.
The `getFinalSizes()` method should be the actual factory method. The data passed to construct should be passed to the method.

That would make this class stateless. Instead the method would act as factory and build the actual array.
That way the rootline can be injected into consumer classes. And the factory can define its own dependencies via constructor.
I'd also state that this is a factory by renaming the class.

Reply:
I think this is already fixed.